src/main/java/uk/gov/pay/webhooks/message/WebhookMessageService.java

```diff
@@ -68,15 +68,15 @@ public void handleInternalEvent(InternalEvent event) throws IOException, Interru
     }
 
     private WebhookMessageEntity buildWebhookMessage(WebhookEntity webhook, InternalEvent event, LedgerTransaction ledgerTransaction) {
-        JsonNode resource = objectMapper.valueToTree(ledgerTransaction); // will probably need some more transformation
+        JsonNode resource = objectMapper.valueToTree(ledgerTransaction); 
 
         var webhookMessageEntity = new WebhookMessageEntity();
         webhookMessageEntity.setExternalId(idGenerator.newExternalId());
         webhookMessageEntity.setCreatedDate(Date.from(instantSource.instant()));
         webhookMessageEntity.setWebhookEntity(webhook);
-        webhookMessageEntity.setEventDate(Date.from(event.eventDate().toInstant()));
+        webhookMessageEntity.setEventDate(Date.from(event.eventDate()));
         webhookMessageEntity.setEventType(eventTypeDao.findByName(EventMapper.getWebhookEventNameFor(event.eventType())).orElseThrow(IllegalArgumentException::new));
-        webhookMessageEntity.setResource(resource);
+        webhookMessageEntity.setResource(objectMapper.valueToTree(WebhookMessage.of(webhookMessageEntity, event, resource)));
```

Comment:
Do we want to presist this formatted payload in the database or do we think it would be better to keep the resource stored separately and format this at the time of sending? I had initially thought the latter to preserve storing just the resource but happy to discuss pros/ cons 

If we did go with this method I think we should rename this property from the entity through to the database to make it clear it's a cache of the webhook payload not just the resource fetched from the event store

Reply:
@sfount As discussed offline, we think it makes sense to persist the individual fields from the wrapper and then do the transform at send time. Maybe I can change this PR to just do the transform at send time, and then do a separate PR to persist the individual wrapper fields?

Reply:
@sfount ...OK I have done this now in this PR in de53ccf